### ISSUE TITLE
Decrypt JWE payload with .p12 file and password

### ISSUE THREAD

**Issue by archana27m** [Spomky-Labs/jose#192]
I am trying to decrypt using code snippet and documentation provided here : https://github.com/Spomky-Labs/jose/blob/master/doc/operation/Decrypt.md

Facing issue as  Fatal error: Uncaught Assert\InvalidArgumentException: Unable to load the key in C:\xampp\htdocs\Encryption\vendor\beberlei\assert\lib\Assert\Assertion.php on line 285

Please let me know what can be the issue or proper example to do it.

Note : I am using PHP 7.0 and using Xampp on windows 7.

**Comment by Spomky** [comment]
Hi @archana27m,

I am afraid but at the moment there is no possibility to directly load PKCS12 certificates.
You have to load it first and then create a JWK object.

Example:

```
use Jose\Factory\JWKFactory;

// Load the PKCS#12 file
$res = openssl_pkcs12_read(file_get_contents('/path/to/your/pkcs.p12'),$certs, 'YOUR SECRET IF NEEDED');
if (false === $res) {
    throw new \RuntimeException('Unable to load the certificates.');
}

// Normally you should have an array with a key "pkey" that represents your private key
$jwk = JWKFactory::createFromKey($certs['pkey']);
```

I think I will add a convenient method.

**Comment by archana27m** [comment]
Hi @Spomky , thanks for the solution. Is it possible for encryption as well using public key?

**Comment by Spomky** [comment]
Great!

I let this issue opened.
That feature will be part of the next major release.

**Comment by archana27m** [comment]
@Spomky,  Just one more thing, is it possible to get json response in decryption, as currently its giving as associative array? Do we need to specify anything for format?

**Comment by Spomky** [comment]
Can you try with the following script ?

```php
$json = (object) $jwe->getPayload();
```

**Comment by archana27m** [comment]
No, It's creating default object with arrays, is there any specific parameter to be passed while decrypting using loader?

**Comment by Spomky** [comment]
OK. Maybe [the accepted answer](https://stackoverflow.com/a/35099716/2157818)?

**Comment by archana27m** [comment]
No, I just wanted to know if it converts into array by default? if we have valid json, it does not return json string?

**Comment by Spomky** [comment]
The v6 of this library always tries to convert the payload.
The next major release will never do that and will always return a string (or null is the payload is detached).

**Comment by Spomky** [comment]
Done. See 8fcef2014f52425e97ba128608c10953c525e0b2.
Will be available on `v7.0.0`.
